view/frontend/web/js/view/payment/method-renderer/omise-offline-tesco-method.js

```diff
@@ -5,15 +5,17 @@ define(
         'Magento_Checkout/js/model/full-screen-loader',
         'Magento_Checkout/js/action/redirect-on-success',
         'Magento_Checkout/js/model/quote',
-        'Magento_Checkout/js/model/error-processor'
+        'Magento_Checkout/js/model/error-processor',
+        'Omise_Payment/js/view/payment/payment-tools'
     ],
     function (
         ko,
         Component,
         fullScreenLoader,
         redirectOnSuccessAction,
         quote,
-        errorProcessor
+        errorProcessor,
+        paymentTools,
```

Comment:
Would most of this refactoring be better achieved using 'mixins'? - would remove the need for all the single line stub functions that just call the same function elsewhere. Also removes the need for `paymentTools` to be passed in.

https://erfanimani.com/three-ways-to-modify-magento-2-javascript

Reply:
how would you do it?
I might misunderstand it, but I see `mixins` being used for other stuff, we are reducing code instead of replacing magento - modules with ours.

Reply:
Well, we have 2 options really:

a) Make a generic payment method renderer module which contains all the common 'methods', then in the individual modules for each payment method, we would bring in that module, and return a modified verison of it (adding the stuff specific to that payment method). You'd do this by modifiying the object directly (prototypes, adding methods etc.)

b) The other way is to do similar to the above, but to use the method already provided in Magento - 'mixins'. Create a mixin with all the common functionality, and mix it in to small modules specific to each individual payment type

Reply:
After reviewing again, it really does seem we should move to some kind of parent-child arrangement

> > The `getCode` function in each of the method renderers is called repeatedly and it just returns a string. Is there any reason why this code cannot just be a constant / property?
> 
> it needs to be a function if you want to access it from `HTML`

This is incorrect

Reply:
thanks @jonrandy, you are welcome to submit your proposal.